dynamic_instrumentation/src/instrument.rs

```diff
@@ -340,35 +340,39 @@ impl<'tcx> MutVisitor<'tcx> for RewriteAddressTakenLocals<'tcx> {
 }
 
 impl<'tcx> Visitor<'tcx> for CollectInstrumentationPoints<'_, 'tcx> {
-    fn visit_place(&mut self, place: &Place<'tcx>, context: PlaceContext, location: Location) {
-        self.super_place(place, context, location);
-
-        let field_fn = self.hooks().find("ptr_field");
-
-        let base_ty = self.local_decls()[place.local].ty;
-
-        // Instrument field projections on raw-ptr places
-        if is_region_or_unsafe_ptr(base_ty) && context.is_use() {
-            for (base, elem) in place.iter_projections() {
-                if let PlaceElem::Field(field, _) = elem {
-                    let proj_dest = || {
-                        // Only the last field projection gets a destination
-                        self.assignment()
-                            .as_ref()
-                            .map(|(dest, _)| dest)
-                            .copied()
-                            .filter(|_| base.projection.len() == place.projection.len() - 1)
-                    };
-                    self.loc(location, location, field_fn)
-                        .arg_var(place.local)
-                        .arg_index_of(field)
-                        .source(place)
-                        .dest_from(proj_dest)
-                        .add_to(self);
-                }
-            }
-        }
-    }
+// TODO(ahomescu): this is completely broken and needs a rewrite
```

Comment:
Hmmm, did I break this all the way back in 2c6b0ff3c0f95c0c4eecd3b02e25bba469b588c9?

Reply:
I don't think so, this problem seems to have always been here in some form or another.